src/controllers/customercontroller.js

```diff
@@ -6,29 +6,81 @@ class CustomerController extends OrderController
             "src/views/template/customer.html",
             "#customers",
             {}
-        );
+        ).done(() => this.registerEvent());
+    }
+
+    registerEvent()
+    {
+        $(document).on('click', '.customer-complete', function(e) {
```

Comment:
In that light, move the `.done()` call from the `view()` method to the `_updateOrderView()`. That way you can use the click function attached to ".customer-complete", rather than binding to the document.